# Data Analysis Challange

No	Challenge <br>
1	Team (teamID )dengan pertandingan home terendah. <br>
2	Nama Team (name) dan Tahun (yearID) dimana Kemenangan (W) terbanyak terjadi. <br>
3	Tahun (yearID) dimana pertandingan terbanyak (G) terjadi, serta jumlah pertandingannya. <br>
4	Stadion (park) dengan total kehadiran (attendance) tertinggi dalam 5 tahun terakhir. <br>
5	Tiga Nama Team (name) dengan Rata-Rata Homeruns(HRA) tertinggi dalam 10 tahun terakhir. <br>

<br>
BONUS Seorang pemain bernama Ben bergabung dengan tim New York Mets selama tahun 2004 hingga 2012, cobalah untuk meneliti bagaimana performa tim pada periode tersebut, bandingkan dengan periode yang sama sebelum dan sesudahnya. Apakah Rekrutmen Ben merupakan hal yang positif pada Tim tersebut?

In [1]:
# Import data
import pandas as pd
Teams = pd.read_csv('https://raw.githubusercontent.com/chadwickbureau/baseballdatabank/master/core/Teams.csv')
Teams.head()

yearID lgID teamID franchID divID  Rank   G  Ghome   W   L  ...  DP     FP  \
0    1871  NaN    BS1      BNA   NaN     3  31    NaN  20  10  ...  24  0.834   
1    1871  NaN    CH1      CNA   NaN     2  28    NaN  19   9  ...  16  0.829   
2    1871  NaN    CL1      CFC   NaN     8  29    NaN  10  19  ...  15  0.818   
3    1871  NaN    FW1      KEK   NaN     7  19    NaN   7  12  ...   8  0.803   
4    1871  NaN    NY2      NNA   NaN     5  33    NaN  16  17  ...  14  0.840   

                      name                          park  attendance  BPF  \
0     Boston Red Stockings           South End Grounds I         NaN  103   
1  Chicago White Stockings       Union Base-Ball Grounds         NaN  104   
2   Cleveland Forest Citys  National Association Grounds         NaN   96   
3     Fort Wayne Kekiongas                Hamilton Field         NaN  101   
4         New York Mutuals      Union Grounds (Brooklyn)         NaN   90   

   PPF  teamIDBR  teamIDlahman45  teamIDretro  
0   98       BOS             BS1          BS1  
1  102       CHI             CH1          CH1  
2  100       CLE             CL1          CL1  
3  107       KEK             FW1          FW1  
4   88       NYU             NY2          NY2  

[5 rows x 48 columns]

### 1. Team (teamID )dengan pertandingan home terendah

#### Cara Cepat

In [2]:
# Mencari TeamID dengan pertandingan home terendah
df_sorted = Teams.sort_values(by='Ghome')
df_sorted

yearID lgID teamID franchID divID  Rank    G  Ghome   W   L  ...   DP  \
2948    2020   AL    SEA      SEA     W     3   60   24.0  27  33  ...   48   
2939    2020   NL    MIA      FLA     E     2   60   26.0  31  29  ...   60   
2953    2020   AL    TOR      TOR     E     3   60   26.0  32  28  ...   47   
2950    2020   NL    SLN      STL     C     3   58   27.0  30  28  ...   46   
2934    2020   AL    DET      DET     C     5   58   27.0  23  35  ...   46   
...      ...  ...    ...      ...   ...   ...  ...    ...  ..  ..  ...  ...   
622     1915   FL    CHF      CHH   NaN     2  155    NaN  86  66  ...  102   
627     1915   FL    KCF      KCP   NaN     4  153    NaN  81  72  ...   96   
628     1915   FL    NEW      NEW   NaN     5  155    NaN  80  72  ...  124   
634     1915   FL    PTF      PBS   NaN     3  156    NaN  86  67  ...   98   
636     1915   FL    SLF      SLI   NaN     1  159    NaN  87  67  ...  111   

         FP                 name               park  attendance  BPF  PPF  \
2948  0.989     Seattle Mariners      T-Mobile Park         0.0   94   95   
2939  0.981        Miami Marlins       Marlins Park         0.0   97   99   
2953  0.982    Toronto Blue Jays       Sahlen Field         0.0  100   99   
2950  0.983  St. Louis Cardinals  Busch Stadium III         0.0   97   96   
2934  0.987       Detroit Tigers      Comerica Park         0.0  102  105   
...     ...                  ...                ...         ...  ...  ...   
622   0.963       Chicago Whales      Wrigley Field         NaN   94   92   
627   0.962  Kansas City Packers                NaN         NaN   97   97   
628   0.963        Newark Pepper                NaN         NaN   94   94   
634   0.971    Pittsburgh Rebels                NaN         NaN   99   99   
636   0.968   St. Louis Terriers                NaN         NaN  105  105   

      teamIDBR  teamIDlahman45  teamIDretro  
2948       SEA             SEA          SEA  
2939       MIA             FLO          MIA  
2953       TOR             TOR          TOR  
2950       STL             SLN          SLN  
2934       DET             DET          DET  
...        ...             ...          ...  
622        CHI             CHF          CHF  
627        KCP             KCF          KCF  
628        NEW             NEW          NEW  
634        PBS             PTF          PTF  
636        SLM             SLF          SLF  

[3015 rows x 48 columns]

#### Mencari team ID dengan pertadingan home terendah

Menggunakan Variabel Berikut : <br>
- teamID -> 	Team; a factor <br>
- Ghome  ->   Games played at home

In [3]:
# Memilih variabel team ID dan Ghome
Data_home = Teams[['teamID', 'Ghome']]

# Membuat DataFrame dari data
df_home = pd.DataFrame(Data_home)
df_home

teamID  Ghome
0       BS1    NaN
1       CH1    NaN
2       CL1    NaN
3       FW1    NaN
4       NY2    NaN
...     ...    ...
3010    SLN   81.0
3011    TBA   81.0
3012    TEX   81.0
3013    TOR   81.0
3014    WAS   81.0

[3015 rows x 2 columns]

In [4]:
## Menghilangkan missing Value

# Menghapus baris yang berisi nilai yang hilang
df_cleaned = df_home.dropna()

# Hasil setelah menghilangkan missing value
data_pertandingan_home = pd.DataFrame(df_cleaned)
data_pertandingan_home

teamID  Ghome
383     BLA   66.0
384     BOS   69.0
385     BRO   68.0
386     BSN   70.0
387     CHA   71.0
...     ...    ...
3010    SLN   81.0
3011    TBA   81.0
3012    TEX   81.0
3013    TOR   81.0
3014    WAS   81.0

[2616 rows x 2 columns]

In [5]:
# Mengambil data teamID 
data_pertandingan_home['teamID'].unique()

array(['BLA', 'BOS', 'BRO', 'BSN', 'CHA', 'CHN', 'CIN', 'CLE', 'DET',
       'MLA', 'NY1', 'PHA', 'PHI', 'PIT', 'SLN', 'WS1', 'SLA', 'NYA',
       'ML1', 'BAL', 'KC1', 'LAN', 'SFN', 'LAA', 'MIN', 'WS2', 'HOU',
       'NYN', 'CAL', 'ATL', 'OAK', 'KCA', 'MON', 'SDN', 'SE1', 'ML4',
       'TEX', 'SEA', 'TOR', 'COL', 'FLO', 'ANA', 'ARI', 'MIL', 'TBA',
       'WAS', 'MIA'], dtype=object)

In [6]:
# Memfilter Team (teamID ) dengan pertandingan home terendah
data_pertandingan_home.sort_values(by='Ghome')

teamID  Ghome
2948    SEA   24.0
2939    MIA   26.0
2953    TOR   26.0
2950    SLN   27.0
2934    DET   27.0
...     ...    ...
2645    PHI   84.0
1545    CHA   84.0
2678    SEA   84.0
2092    PIT   84.0
1482    CHN   84.0

[2616 rows x 2 columns]

### Interpretasi

Berdasarkan hasil yang ditampilkan, dapat diketahui bahwa Team (teamID ) dengan pertandingan home terendah yaitu dengan teamID **"SEA"** dengan nilai Ghome sebesar **24.0**

### 2. Nama Team (name) dan Tahun (yearID) dimana Kemenangan (W) terbanyak terjadi.

Menggunakan Variabel Berikut : <br>
- name	    -> Team’s full name <br>
- yearID	-> Year <br>
- W	        -> Wins

In [7]:
# Mengelompokan variabel yang dibutuhkan
data_team = Teams[['name', 'yearID', 'W']]
df_team = pd.DataFrame(data_team)
df_team

name  yearID   W
0        Boston Red Stockings    1871  20
1     Chicago White Stockings    1871  19
2      Cleveland Forest Citys    1871  10
3        Fort Wayne Kekiongas    1871   7
4            New York Mutuals    1871  16
...                       ...     ...  ..
3010      St. Louis Cardinals    2022  93
3011           Tampa Bay Rays    2022  86
3012            Texas Rangers    2022  68
3013        Toronto Blue Jays    2022  92
3014     Washington Nationals    2022  55

[3015 rows x 3 columns]

In [8]:
# Mengelompokkan data berdasarkan 'name' dan 'yearID', kemudian menjumlahkan kemenangan (W)
grouped = df_team.groupby(['name', 'yearID'])['W'].sum().reset_index()
grouped

name  yearID   W
0     Altoona Mountain City    1884   6
1            Anaheim Angels    1997  84
2            Anaheim Angels    1998  85
3            Anaheim Angels    1999  70
4            Anaheim Angels    2000  82
...                     ...     ...  ..
3007   Washington Statesmen    1891  44
3008  Wilmington Quicksteps    1884   2
3009    Worcester Ruby Legs    1880  40
3010    Worcester Ruby Legs    1881  32
3011    Worcester Ruby Legs    1882  18

[3012 rows x 3 columns]

In [9]:
# Menemukan baris dengan kemenangan terbanyak
baris_terbanyak = grouped[grouped['W'] == grouped['W'].max()]
baris_terbanyak

name  yearID    W
1711  New York Giants    1890  137

In [10]:
# Menampilkan hasil
print("Nama Team (name) dengan Kemenangan (win) terbanyak:", baris_terbanyak['name'].values[0])
print("Tahun (yearID) dimana Kemenangan (win) terbanyak terjadi:", baris_terbanyak['yearID'].values[0])
print("Jumlah Kemenangan terbanyak:", baris_terbanyak['W'].values[0])

Nama Team (name) dengan Kemenangan (win) terbanyak: New York Giants
Tahun (yearID) dimana Kemenangan (win) terbanyak terjadi: 1890
Jumlah Kemenangan terbanyak: 137


#### Interpretasi

Berdasarkan hasil di atas, dapat diketahui bahwa Nama Team (name) dan Tahun (yearID) dimana Kemenangan (W) terbanyak terjadi yaitu **New York Giants** yang terjadi pada **tahun 1890** dengan kemenangan sebanyak **137 kemenangan**.

### 3 Tahun (yearID) dimana pertandingan terbanyak (G) terjadi, serta jumlah pertandingannya

Menggunakan Variabel Berikut : <br>
- yearID ->	Year <br>
- G	     -> Games played

In [11]:
# Mengelompokan variabel yang dibutuhkan
data_pertandingan = Teams[['yearID', 'G']]
df_pertandingan = pd.DataFrame(data_pertandingan)
df_pertandingan

yearID    G
0       1871   31
1       1871   28
2       1871   29
3       1871   19
4       1871   33
...      ...  ...
3010    2022  162
3011    2022  162
3012    2022  162
3013    2022  162
3014    2022  162

[3015 rows x 2 columns]

In [12]:
# Mengelompokkan data berdasarkan 'tahun', kemudian menjumlahkan pertandingan
grouped_match = df_pertandingan.groupby('yearID')['G'].sum().reset_index()
grouped_match

yearID     G
0      1871   254
1      1872   366
2      1873   398
3      1874   464
4      1875   690
..      ...   ...
147    2018  4862
148    2019  4858
149    2020  1796
150    2021  4858
151    2022  4860

[152 rows x 2 columns]

In [13]:
# Menemukan tahun dengan pertandingan terbanyak
tahun_terbanyak = grouped_match[grouped_match['G'] == grouped_match['G'].max()]
tahun_terbanyak

yearID     G
127    1998  4864

In [14]:
# Menampilkan hasil
print("Tahun (yearID) dengan pertandingan terbanyak:", tahun_terbanyak['yearID'].values[0])
print("Jumlah pertandingan terbanyak:", tahun_terbanyak['G'].values[0])

Tahun (yearID) dengan pertandingan terbanyak: 1998
Jumlah pertandingan terbanyak: 4864


### Interpretasi

Berdasarkan hasil di atas, dapat diketahui bahwa Tahun (yearID) dimana pertandingan terbanyak (G) terjadi, serta jumlah pertandingannya yaitu pada **tahun 1998** dengan jumlah pertandingannya **sebanyak 4864**.

### 4. Stadion (park) dengan total kehadiran (attendance) tertinggi dalam 5 tahun terakhir.

Menggunakan Variabel Berikut : <br>
- park	     -> Name of team’s home ballpark <br>
- yearID     ->	Year <br>
- attendance ->	Home attendance total

In [15]:
# Mengelompokan variabel yang dibutuhkan
data_stadion = Teams[['park', 'attendance', 'yearID']]
df_stadion = pd.DataFrame(data_stadion)
df_stadion

park  attendance  yearID
0              South End Grounds I         NaN    1871
1          Union Base-Ball Grounds         NaN    1871
2     National Association Grounds         NaN    1871
3                   Hamilton Field         NaN    1871
4         Union Grounds (Brooklyn)         NaN    1871
...                            ...         ...     ...
3010             Busch Stadium III   3320551.0    2022
3011               Tropicana Field   1128127.0    2022
3012              Globe Life Field   2011361.0    2022
3013                 Rogers Centre   2653830.0    2022
3014                Nationals Park   2026401.0    2022

[3015 rows x 3 columns]

In [16]:
# Mengelompokan stadion 5 tahun terakhir
tahun_terakhir = df_stadion[df_stadion['yearID'] >= 2018]
tahun_terakhir

park  attendance  yearID
2865                  Chase Field   2242695.0    2018
2866                SunTrust Park   2555781.0    2018
2867  Oriole Park at Camden Yards   1564192.0    2018
2868               Fenway Park II   2895575.0    2018
2869        Guaranteed Rate Field   1608817.0    2018
...                           ...         ...     ...
3010            Busch Stadium III   3320551.0    2022
3011              Tropicana Field   1128127.0    2022
3012             Globe Life Field   2011361.0    2022
3013                Rogers Centre   2653830.0    2022
3014               Nationals Park   2026401.0    2022

[150 rows x 3 columns]

In [17]:
# Mengelompokkan data berdasarkan 'nama_std' dan menjumlahkan pertandingan
grouped_std = tahun_terakhir.groupby('park')['attendance'].sum().reset_index()
grouped_std

park  attendance
0                      AT&T Park   3156185.0
1       Angel Stadium of Anaheim  10012378.0
2              Busch Stadium III  12307061.0
3                    Chase Field   7026414.0
4                     Citi Field   8744190.0
5             Citizens Bank Park   8678171.0
6                  Comerica Park   6036565.0
7                    Coors Field  10545197.0
8                 Dodger Stadium  14497910.0
9                 Fenway Park II  10170614.0
10              Globe Life Field   4121619.0
11  Globe Life Park in Arlington   4240101.0
12      Great American Ball Park   6338835.0
13         Guaranteed Rate Field   6864336.0
14              Kauffman Stadium   5582065.0
15                  Marlins Park   3172510.0
16                   Miller Park  10020910.0
17              Minute Maid Park  10595423.0
18                Nationals Park   8281329.0
19                 O.co Coliseum   4725159.0
20                   Oracle Park   6869930.0
21   Oriole Park at Camden Yards   5033595.0
22                      PNC Park   5073711.0
23                    Petco Park   9744355.0
24             Progressive Field   6075581.0
25                 Rogers Centre   7535126.0
26                  Safeco Field   2299489.0
27                  Sahlen Field         0.0
28                 SunTrust Park  10641059.0
29                 T-Mobile Park   5295115.0
30                  Target Field   7364676.0
31               Tropicana Field   4222907.0
32                 Wrigley Field  10871668.0
33            Yankee Stadium III  11883320.0

In [18]:
# Menemukan stadion dengan total kehadiran tertinggi
stadion_tertinggi = grouped_std[grouped_std['attendance'] == grouped_std['attendance'].max()]
stadion_tertinggi

park  attendance
8  Dodger Stadium  14497910.0

In [19]:
# Menampilkan hasil
print("Stadion (park) dengan total kehadiran tertinggi dalam 2 tahun terakhir:", stadion_tertinggi['park'].values[0])
print("Total kehadiran tertinggi:", stadion_tertinggi['attendance'].values[0])

Stadion (park) dengan total kehadiran tertinggi dalam 2 tahun terakhir: Dodger Stadium
Total kehadiran tertinggi: 14497910.0


### Interpretasi

Berdasarkan hasil diatas, dapat diketahui bahwa Stadion (park) dengan total kehadiran (attendance) tertinggi dalam 5 tahun terakhir adalah **Dodger Stadium** dengan **jumlah kehadirannya yaitu 14497910.0**

### 5 Tiga Nama Team (name) dengan Rata-Rata Homeruns(HRA) tertinggi dalam 10 tahun terakhir.

Menggunakan Variabel Berikut : <br>
- name	 -> Team’s full name <br>
- yearID -> Year <br>
- HRA	 -> Homeruns allowed 

In [20]:
# Mengelompokan variabel yang dibutuhkan
data_HRA = Teams[['name', 'HRA', 'yearID']]
df_HRA = pd.DataFrame(data_HRA)
df_HRA

name  HRA  yearID
0        Boston Red Stockings    2    1871
1     Chicago White Stockings    6    1871
2      Cleveland Forest Citys   13    1871
3        Fort Wayne Kekiongas    5    1871
4            New York Mutuals    7    1871
...                       ...  ...     ...
3010      St. Louis Cardinals  146    2022
3011           Tampa Bay Rays  172    2022
3012            Texas Rangers  169    2022
3013        Toronto Blue Jays  180    2022
3014     Washington Nationals  244    2022

[3015 rows x 3 columns]

In [21]:
# Mengambil data hanya untuk 10 tahun terakhir (tahun 2013-2022)
tahun_terakhir = df_HRA[df_HRA['yearID'] >= 2013]
tahun_terakhir

name  HRA  yearID
2715  Arizona Diamondbacks  176    2013
2716        Atlanta Braves  127    2013
2717     Baltimore Orioles  202    2013
2718        Boston Red Sox  156    2013
2719     Chicago White Sox  182    2013
...                    ...  ...     ...
3010   St. Louis Cardinals  146    2022
3011        Tampa Bay Rays  172    2022
3012         Texas Rangers  169    2022
3013     Toronto Blue Jays  180    2022
3014  Washington Nationals  244    2022

[300 rows x 3 columns]

In [22]:
# Mengelompokkan data berdasarkan 'team' dan menghitung rata-rata HRA
grouped_HRA = tahun_terakhir.groupby('name')['HRA'].mean().reset_index()
grouped_HRA

name         HRA
0            Arizona Diamondbacks  179.500000
1                  Atlanta Braves  154.300000
2               Baltimore Orioles  199.900000
3                  Boston Red Sox  170.900000
4                    Chicago Cubs  163.400000
5               Chicago White Sox  176.400000
6                 Cincinnati Reds  194.400000
7             Cleveland Guardians  172.000000
8               Cleveland Indians  164.777778
9                Colorado Rockies  178.000000
10                 Detroit Tigers  177.100000
11                 Houston Astros  162.400000
12             Kansas City Royals  170.400000
13  Los Angeles Angels of Anaheim  180.100000
14            Los Angeles Dodgers  150.600000
15                  Miami Marlins  156.600000
16              Milwaukee Brewers  170.400000
17                Minnesota Twins  180.400000
18                  New York Mets  164.600000
19               New York Yankees  178.400000
20              Oakland Athletics  171.700000
21          Philadelphia Phillies  177.000000
22             Pittsburgh Pirates  157.300000
23               San Diego Padres  170.100000
24           San Francisco Giants  150.800000
25               Seattle Mariners  185.900000
26            St. Louis Cardinals  140.200000
27                 Tampa Bay Rays  164.700000
28                  Texas Rangers  184.800000
29              Toronto Blue Jays  181.100000
30           Washington Nationals  172.600000

In [23]:
# Mengurutkan DataFrame berdasarkan rata-rata HRA secara menurun
sorted_df = grouped_HRA.sort_values(by='HRA', ascending=False)
sorted_df

name         HRA
2               Baltimore Orioles  199.900000
6                 Cincinnati Reds  194.400000
25               Seattle Mariners  185.900000
28                  Texas Rangers  184.800000
29              Toronto Blue Jays  181.100000
17                Minnesota Twins  180.400000
13  Los Angeles Angels of Anaheim  180.100000
0            Arizona Diamondbacks  179.500000
19               New York Yankees  178.400000
9                Colorado Rockies  178.000000
10                 Detroit Tigers  177.100000
21          Philadelphia Phillies  177.000000
5               Chicago White Sox  176.400000
30           Washington Nationals  172.600000
7             Cleveland Guardians  172.000000
20              Oakland Athletics  171.700000
3                  Boston Red Sox  170.900000
16              Milwaukee Brewers  170.400000
12             Kansas City Royals  170.400000
23               San Diego Padres  170.100000
8               Cleveland Indians  164.777778
27                 Tampa Bay Rays  164.700000
18                  New York Mets  164.600000
4                    Chicago Cubs  163.400000
11                 Houston Astros  162.400000
22             Pittsburgh Pirates  157.300000
15                  Miami Marlins  156.600000
1                  Atlanta Braves  154.300000
24           San Francisco Giants  150.800000
14            Los Angeles Dodgers  150.600000
26            St. Louis Cardinals  140.200000

In [26]:
# Mengambil tiga nama tim dengan rata-rata HRA tertinggi
tiga_tertinggi = sorted_df.head(3)
tiga_tertinggi

name    HRA
2   Baltimore Orioles  199.9
6     Cincinnati Reds  194.4
25   Seattle Mariners  185.9

In [27]:
print("Tiga Nama Team (name) dengan Rata-Rata Homeruns (HRA) tertinggi dalam 10 tahun terakhir:")
print(tiga_tertinggi)

Tiga Nama Team (name) dengan Rata-Rata Homeruns (HRA) tertinggi dalam 10 tahun terakhir:
                 name    HRA
2   Baltimore Orioles  199.9
6     Cincinnati Reds  194.4
25   Seattle Mariners  185.9


### Interpretasi

Berdasarkan hasil di atas, dapat diketahui bahwa Tiga Nama Team (name) dengan Rata-Rata Homeruns(HRA) tertinggi dalam 10 tahun terakhir yaitu <br>
**1. Baltimore Orioles dengan HRA 199.9,** <br> 
**2. Cincinnati Reds dengan HRA  194.4,** <br>
**3. Seattle Mariners dengan HRA 185.9.**

### BONUS 

Seorang pemain bernama Ben bergabung dengan tim New York Mets selama tahun 2004 hingga 2012, cobalah untuk meneliti bagaimana performa tim pada periode tersebut, bandingkan dengan periode yang sama sebelum dan sesudahnya. Apakah Rekrutmen Ben merupakan hal yang positif pada Tim tersebut?

Pada kasus ini Menggunakan data <br>
- name	-> Team’s full name <br>
- yearID ->	Year
- W	    -> Wins <br>
- L	    -> Losses <br>
- LgWin	-> League Champion(Y or N)

Win dan Losses yang digunakan adalah rata-rata kemenangan dan kekalahan dari tim New York Mets selama tahun 2004 hingga 2012 dan dibandingkan sebelum dan setelah Ben bergabung, Karena hal tersebut menunjukan performa tim selama ben bergabung apakah berdampak terhadap pertandingan yang dilakukan oleh tim. <br>

Untuk LgWin digunakan jumlah juara yang diperoleh selama Ben bergabung di tim New York Mets selama tahun 2004 hingga 2012 dan dibandingkan sebelum dan setelah Ben bergabung. Hal tersebut menunjukan performa tim secara keseluruhan dengan mendapatkan trophy di akhir liga. 

In [28]:
# Mengelompokan hanya variabel yang diinginkan
data_bonus = Teams[['name', 'W', 'L', 'yearID', 'LgWin']]
df_bonus = pd.DataFrame(data_bonus)
df_bonus

name   W    L  yearID LgWin
0        Boston Red Stockings  20   10    1871     N
1     Chicago White Stockings  19    9    1871     N
2      Cleveland Forest Citys  10   19    1871     N
3        Fort Wayne Kekiongas   7   12    1871     N
4            New York Mutuals  16   17    1871     N
...                       ...  ..  ...     ...   ...
3010      St. Louis Cardinals  93   69    2022     N
3011           Tampa Bay Rays  86   76    2022     N
3012            Texas Rangers  68   94    2022     N
3013        Toronto Blue Jays  92   70    2022     N
3014     Washington Nationals  55  107    2022     N

[3015 rows x 5 columns]

In [29]:
# Memfilter data untuk tim dan tahun 2004-2012
data_tim_2004_2012 = df_bonus[(df_bonus['name'] == 'New York Mets') & (df_bonus['yearID'] >= 2004) & (df_bonus['yearID'] <= 2012)]
data_tim_2004_2012

name   W   L  yearID LgWin
2464  New York Mets  71  91    2004     N
2493  New York Mets  83  79    2005     N
2523  New York Mets  97  65    2006     N
2553  New York Mets  88  74    2007     N
2583  New York Mets  89  73    2008     N
2613  New York Mets  70  92    2009     N
2643  New York Mets  79  83    2010     N
2673  New York Mets  77  85    2011     N
2703  New York Mets  74  88    2012     N

In [30]:
# Menghitung jumlah kemenangan selama 2004-2012
jumlah_win_2004_2012 = data_tim_2004_2012['W'].sum()
jumlah_win_2004_2012

728

In [31]:
# Menghitung jumlah kekalahan selama 2004-2012
jumlah_loss_2004_2012 = data_tim_2004_2012['L'].sum()
jumlah_loss_2004_2012

730

In [33]:
# Menghitung rata-rata kemenangan  2004-2012
rata_rata_win_2004_2012 = data_tim_2004_2012['W'].mean()
rata_rata_win_2004_2012

80.88888888888889

In [35]:
# Menghitung rata-rata kekalahan  2004-2012
rata_rata_loss_2004_2012 = data_tim_2004_2012['L'].mean()
rata_rata_loss_2004_2012

81.11111111111111

In [58]:
# Menghitung jumlah berapa kali tim menjadi juara 
jumlah_juara = len(data_tim_2004_2012[(data_tim_2004_2012['name'] == 'New York Mets') & (data_tim_2004_2012['LgWin'] == 'Y')])
jumlah_juara

0

In [38]:
# Memfilter data pada tahun sebelum 2004
data_sebelum_2004 = df_bonus[(df_bonus['name'] == 'New York Mets') & (df_bonus['yearID'] < 2004)]
data_sebelum_2004

name    W    L  yearID LgWin
1391  New York Mets   40  120    1962     N
1411  New York Mets   51  111    1963     N
1431  New York Mets   53  109    1964     N
1451  New York Mets   50  112    1965     N
1471  New York Mets   66   95    1966     N
1491  New York Mets   61  101    1967     N
1510  New York Mets   73   89    1968     N
1532  New York Mets  100   62    1969     Y
1557  New York Mets   83   79    1970     N
1581  New York Mets   83   79    1971     N
1605  New York Mets   83   73    1972     N
1629  New York Mets   82   79    1973     Y
1653  New York Mets   71   91    1974     N
1677  New York Mets   82   80    1975     N
1701  New York Mets   86   76    1976     N
1725  New York Mets   64   98    1977     N
1751  New York Mets   66   96    1978     N
1777  New York Mets   63   99    1979     N
1803  New York Mets   67   95    1980     N
1829  New York Mets   41   62    1981     N
1855  New York Mets   65   97    1982     N
1881  New York Mets   68   94    1983     N
1907  New York Mets   90   72    1984     N
1933  New York Mets   98   64    1985     N
1959  New York Mets  108   54    1986     Y
1985  New York Mets   92   70    1987     N
2011  New York Mets  100   60    1988     N
2037  New York Mets   87   75    1989     N
2063  New York Mets   91   71    1990     N
2089  New York Mets   77   84    1991     N
2115  New York Mets   72   90    1992     N
2143  New York Mets   59  103    1993     N
2171  New York Mets   55   58    1994   NaN
2199  New York Mets   69   75    1995     N
2227  New York Mets   71   91    1996     N
2255  New York Mets   88   74    1997     N
2284  New York Mets   88   74    1998     N
2314  New York Mets   97   66    1999     N
2344  New York Mets   94   68    2000     Y
2374  New York Mets   82   80    2001     N
2404  New York Mets   75   86    2002     N
2434  New York Mets   66   95    2003     N

In [39]:
# Menghitung jumlah kemenangan sebelum 2004
jumlah_win_sebelum_2004 = data_sebelum_2004['W'].sum()
jumlah_win_sebelum_2004

3157

In [40]:
# Menghitung jumlah kekalahan sebelum 2004
jumlah_loss_sebelum_2004 = data_sebelum_2004['L'].sum()
jumlah_loss_sebelum_2004

3507

In [42]:
# Menghitung rata-rata kemenangan sebelum 2004
rata_rata_win_sebelum_2004 = data_sebelum_2004['W'].mean()
rata_rata_win_sebelum_2004

75.16666666666667

In [44]:
# Menghitung rata-rata kekalahan sebelum 2004
rata_rata_loss_sebelum_2004 = data_sebelum_2004['L'].mean()
rata_rata_loss_sebelum_2004

83.5

In [57]:
# Menghitung jumlah kali tim  menjadi juara sebelum 2004
jumlah_juara2 = len(data_sebelum_2004[(data_sebelum_2004['name'] == 'New York Mets') & (data_sebelum_2004['LgWin'] == 'Y')])
jumlah_juara2

4

In [47]:
# Memfilter data pada tahun setelah 2012
data_setelah_2012 = df_bonus[(df_bonus['name'] == 'New York Mets') & (df_bonus['yearID'] > 2012)]
data_setelah_2012

name    W   L  yearID LgWin
2733  New York Mets   74  88    2013     N
2763  New York Mets   79  83    2014     N
2793  New York Mets   90  72    2015     Y
2823  New York Mets   87  75    2016     N
2853  New York Mets   70  92    2017     N
2883  New York Mets   77  85    2018     N
2913  New York Mets   86  76    2019     N
2943  New York Mets   26  34    2020     N
2973  New York Mets   77  85    2021     N
3003  New York Mets  101  61    2022     N

In [48]:
# Menghitung jumlah kemenangan setelah 2012
jumlah_win_setelah_2012 = data_setelah_2012['W'].sum()
jumlah_win_setelah_2012

767

In [49]:
# Menghitung jumlah kekalahan setelah_2012
jumlah_loss_setelah_2012= data_setelah_2012['L'].sum()
jumlah_loss_setelah_2012

751

In [50]:
# Menghitung rata-rata kemenangan setelah_2012
rata_rata_win_setelah_2012 = data_setelah_2012['W'].mean()
rata_rata_win_setelah_2012

76.7

In [51]:
# Menghitung rata-rata kekalahan setelah_2012
rata_rata_loss_setelah_2012 = data_setelah_2012['L'].mean()
rata_rata_loss_setelah_2012

75.1

In [52]:
# Menghitung jumlah kali tim  menjadi juara sebelum 2004
jumlah_juara3 = len(data_setelah_2012[(data_setelah_2012['name'] == 'New York Mets') & (data_setelah_2012['LgWin'] == 'Y')])
jumlah_juara3

1

In [59]:
# Statistik akhir 2004 - 2012
print("Rata-rata kemenangan (W) tim 'New York Mets' dari tahun 2004-20012:", rata_rata_win_2004_2012,
      ", Dan rata-rata kekalahan (L) tim 'New York Mets' dari tahun 2004-2012:", rata_rata_loss_2004_2012, 
     "dan mendapatkan juara sebanyak:", jumlah_juara)

Rata-rata kemenangan (win) tim 'New York Mets' dari tahun 2004-20012: 80.88888888888889 , Dan rata-rata kekalahan (loss) tim 'New York Mets' dari tahun 2004-2012: 81.11111111111111 dan mendapatkan juara sebanyak: 0


In [60]:
# Statistik akhir sebelum 2004
print("Rata-rata kemenangan (W) tim 'New York Mets' sebelum 2004:", rata_rata_win_sebelum_2004,
      ", Dan rata-rata kekalahan (L) tim 'New York Mets' sebelum 2004:", rata_rata_loss_sebelum_2004, 
     "dan mendapatkan juara sebanyak:", jumlah_juara2)

Rata-rata kemenangan (W) tim 'New York Mets' sebelum 2004: 75.16666666666667 , Dan rata-rata kekalahan (L) tim 'New York Mets' sebelum 2004: 83.5 dan mendapatkan juara sebanyak: 4


In [61]:
# Statistik akhir sebelum 2004
print("Rata-rata kemenangan (W) tim 'New York Mets' setelah 2012:", rata_rata_win_setelah_2012,
      ", Dan rata-rata kekalahan (L) tim 'New York Mets' setelah 2012:", rata_rata_loss_setelah_2012, 
     "dan mendapatkan juara sebanyak:", jumlah_juara3)

Rata-rata kemenangan (W) tim 'New York Mets' setelah 2012: 76.7 , Dan rata-rata kekalahan (L) tim 'New York Mets' setelah 2012: 75.1 dan mendapatkan juara sebanyak: 1
